assets/js/apps/search/components/ResultSection.jsx

```diff
@@ -0,0 +1,285 @@
+                <tr>
+                    <th></th>
+                    <th>Name</th>
+                    <th>Size</th>
+                </tr>
+            </thead>
+            <tbody>
+                {body}
+            </tbody>
+        </Table>
+    )
+}
+
+PureResultList.propTypes = {
+    results: PropTypes.arrayOf(Object),
+    error: PropTypes.string,
+    isLoading: PropTypes.bool,
+    selectedItem: PropTypes.number,
+    onItemSelect: PropTypes.func
+}
+
+export function PureResultSection({ resultSets, selectedType,
+    onSelectType, selectedResult, onSelectResult, isLoading, error }) {
+    let counts;
+    if (!resultSets) {
+        resultSets = {};
+        counts = {
+            project: null,
+            experiment: null,
+            dataset: null,
+            datafile: null
+        }
+    } else {
+        counts = {};
+        for (let key in resultSets) {
+            counts[key] = resultSets[key].length;
+        }
+    }
+
+        let selectedEntry = getSelectedEntry(resultSets, selectedResult, selectedType);
+
+        const currentResultSet = resultSets[selectedType],
+            currentCount = counts[selectedType];
+        return (
+            <>
+                <ResultTabs counts={counts} selectedType={selectedType} onChange={onSelectType} />
+                <div role="tabpanel" className="result-section--tabpanel">
+                    {(!isLoading && !error) &&
+                        <p className="result-section--count-summary">
+                            <span>Showing {currentCount} {currentCount > 1 ? "results" : "result"}.</span>
+                        </p>
+                    }
+                    <div className="tabpanel__container--horizontal">
+                        <PureResultList results={currentResultSet} selectedItem={selectedResult} onItemSelect={onSelectResult} isLoading={isLoading} error={error} />
+                        {(!isLoading && !error && currentCount > 0) &&
+                            <EntryPreviewCard
+                                data={selectedEntry}
+                            />
+                        }
+                    </div>
+                </div>
+            </>
+        )
+    }
+
+
+/**
+ * Returns the data of the selected row. Returns null if it cannot get find the selected result.
+ * @param {*} resultSets 
+ * @param {*} selectedResult 
+ * @param {*} selectedType 
+ */
+function getSelectedEntry(resultSets, selectedResult, selectedType) {
+    let selectedEntry = null;
+    if (resultSets && selectedResult) {
+        selectedEntry = resultSets[selectedType].filter(result => result.id === selectedResult)[0];
+    }
+    return selectedEntry;
+}
+
+    export default function ResultSection() {
+        const selectedType = useSelector(state => state.search.selectedType),
+            selectedResult = useSelector(state => state.search.selectedResult),
+            dispatch = useDispatch(),
+            onSelectType = (type) => {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 12.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933948)
```suggestion
        onSelectType = (type) => {
```